Formula/gpgme.rb

```diff
@@ -12,6 +12,8 @@ class Gpgme < Formula
     sha256 "4a4c3e90175b02ded974a8a385b6d9799bcc2873c5a1087c558f350eb103934d" => :yosemite
   end
 
+  patch :DATA
```

Comment:
The patch is needed in order to compile the c++ sources. Without the patch, it fails due to `strdup()` not being defined. The patch is based on the fact that gpgme-1.7.0 does not #include "config.h" in those sources and does compile w/o error.

That said, I've just emailed the gnupg-devel list with a proper solution which involves changes to configure.ac to define `_XOPEN_SOURCE` to `700` instead of `500`.